### DISCUSSION TITLE
consult-kmacro requires ARG as function parameter

### DISCUSSION THREAD

**Discussion by EGmux** [minad/consult#1110] [Q&A] [answered]
- I'm not sure if this intentional or a bug, but the documentation of consult-kmacro seems poorly-worded:

*(consult-kmacro ARG)*

Run a chosen keyboard macro.

**With prefix ARG**, run the macro that many times.
Macros containing mouse clicks are omitted.

- Shouldn't the signature be instead:

*(consult-kmacro &optional ARG)*

I would expect *with prefix* to be optional behavior and not mandatory as the current signature implies.

**Comment by minad** [comment, marked as answer]
> I would expect with prefix to be optional behavior and not mandatory as the current signature implies.

It doesn't really matter. If you invoke the command interactively, the argument will be filled by the `interactive` specification. If you invoke the function programmatically, you have to pass the argument , e.g., 1.
